### PR TITLE
Fix the icom listener test error

### PR DESCRIPTION
**Pull request by laser47-hue**
with regards to the failing test_icom_cli, it looks like the test icom listener is being terminated immediately after the data has finished sending from the mock icom server, which means there is not enough time for the byte stream to be saved to txt files and stored in a temp directory. An easy slightly ugly fix, but i think robust enough is to put a time.sleep(1) between icom_server_process.join() and icom_listen_cli.terminate()

### CODE REVIEW

**Comment by sjswerdloff** [comment]
The pre commit test failed, and it was a formatting issue.  Please take a closer look at the developer instructions regarding installation of precommit hooks and the use of the package 'black' for automatic formatting.  There can be some interaction with the isort tool, in which case there are settings to ensure that they work together properly. 
If you need further assistance please provide details where things got difficult.

**Comment by sjswerdloff** [comment]
@SimonBiggs , the only failing test is for 'pyright' and that isn't "required".
I'm OK with the code (I'll put specific comments in to have a discussion of what might be more optimal, but I'm a firm believer in good working code vs. waiting for optimal).
Once I have approved the *code*, I'm going to leave the *merge* to you (or any other maintainer) or at least wait to hear from you (or another maintainer) on whether to proceed with the merge.

**Review by sjswerdloff — APPROVED** [review]
While putting in a bit of time to sleep and wait for an asynchronous task to complete seems to work ok, is it possible to have some kind of "await" on the asynchronous task given that we can't really proceed until it is done?  Or perhaps have a specific "wait for the data we need to be available" blocking function call that throws an error on timeout (and put the sleep in there and check for whatever the real condition is)?

**Comment by SimonBiggs** [comment]
> I'm OK with the code (I'll put specific comments in to have a discussion of what might be more optimal, but I'm a firm believer in good working code vs. waiting for optimal).

Yup, me too. Thank you Marcus! :)

**Comment by sjswerdloff** [comment]
> While putting in a bit of time to sleep and wait for an asynchronous task to complete seems to work ok, is it possible to have some kind of "await" on the asynchronous task given that we can't really proceed until it is done? Or perhaps have a specific "wait for the data we need to be available" blocking function call that throws an error on timeout (and put the sleep in there and check for whatever the real condition is)?

It appears that the current solution is not sufficient, the same test failed in: #1826

```
            live_files = list(pathlib.Path(temp_dir).glob("**/*.txt"))
>           assert len(live_files) == 256
E           AssertionError: assert 160 == 256
E            +  where 160 = len([PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/165.txt')
```

The above test that failed looks like a good way to check on whether the download is in fact complete and "good to go", so I'll try to take a look at creating a PR that provides a blocking function that has a long, but not way too long timeout (with incremental sleeps of 1 second so we don't waste time either), unless someone else gets to it sooner.

**Comment by laser47-hue** [comment]
@sjswerdloff, it looks like the failure in #1826 wasn't using the new test:
**From the error message:** (doesn't have the time.sleep(1) in it)

            icom_server_process.join()
            icom_listen_cli.terminate()

            live_files = list(pathlib.Path(temp_dir).glob("**/*.txt"))
            assert len(live_files) == 256
E           AssertionError: assert 160 == 256
E            +  where 160 = len([PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/165.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/172.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/111.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/198.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/195.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/068.txt'), ...])


But agreed, your method does sound better.

**Comment by sjswerdloff** [comment]
> @sjswerdloff, it looks like the failure in #1826 wasn't using the new test: **From the error message:** (doesn't have the time.sleep(1) in it)
> 
> ```
>         icom_server_process.join()
>         icom_listen_cli.terminate()
> 
>         live_files = list(pathlib.Path(temp_dir).glob("**/*.txt"))
>         assert len(live_files) == 256
> ```
> 
> E AssertionError: assert 160 == 256 E + where 160 = len([PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/165.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/172.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/111.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/198.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/195.txt'), PosixPath('/tmp/tmp0fapjhjq/live/127.0.0.1/068.txt'), ...])
> 
> But agreed, your method does sound better.

I'm not sure why it wouldn't have used the current main branch as it's starting point.
Thank you for pointing out that it wasn't using the latest code.
I'll try re-running the entire workflow.
And investigate what is needed to ensure the updated main becomes the baseline for the PR.
I'm less focused on "better" and more focused on "will get us through all of our tests" for the moment.
I'm a fan of "good enough" and working.

**Comment by sjswerdloff** [comment]
Seems to have worked itself out, so I'm going to leave well enough alone.
Thank you, again, @laser47-hue for your fix and for identifying root cause of the other build still failing.

**Comment by Matthew-Jennings** [comment]
Agreed, thanks @laser47-hue ! And @sjswerdloff for supporting and reviewing.

**Comment by SimonBiggs** [comment]
@laser47-hue, did you want to add yourself to the active contributors list in time for the new release? 🙂

**Comment by SimonBiggs** [comment]
Also feel free to pull from the README.rst on this pymedphys version in order to pull in an RCCC logo:
https://github.com/pymedphys/pymedphys/tree/v0.37.0

**Comment by laser47-hue** [comment]
@SimonBiggs Sure, that would be great. I'll get something sorted now.

**Pull request merged**
